### PR TITLE
Add dynamic twitter content to candidate page, rewrite candidate page

### PR DESCRIPTION
**Pull request by ganewman**
… to work using templates.

### What is a quick description of the change?

This PR displays the twitter feed for a given candidate on the candidate page using the Twitter for Website Timeline widget.
The candidate's twitter handle is passed along with other official candidate info.

This PR also reorganizes many of the functions in the `script.js` file so that candidate information is displayed on the candidate page using AngularJS templates. Information is passed through the candidate page's `scope`, rather than being inserted using JavaScript. 

This same process will eventually be carried out for the directory page as well.

### Is this fixing an issue?

none

### Are there more details that are relevant?

none

### Check lists (check `x` in `[ ]` of list items)

- [  ] Test written/updating
- [ ] Tests passing
- [X] Coding style (indentation, etc)

Please explain why any are not present, if any.

### CODE REVIEW

File: project/src/main/webapp/candidate-page/candidate-page.component.js [hunk 1]
```diff
@@ -17,9 +17,53 @@
...
 });
+
+/**
+ * Adds candidate information to the page's scope so it can be
+ * inserted by the page template.
+ */
+async function getCandidateInformation(scope) {
+  const candidateId = location.search.substring(
+      location.search.indexOf('=') + 1, location.search.indexOf('&'));
+
```
Comment (KathyFeiyang) [inline project/src/main/webapp/candidate-page/candidate-page.component.js]: nit: remove this empty line

File: project/src/main/webapp/candidate-page/candidate-page.component.js [hunk 2]
```diff
@@ -17,9 +17,53 @@
...
+function addCandidateTwitterInfo(scope, twitterHandle) {
+  scope.twitterHandle = twitterHandle;
+  scope.twitterURL = "https://twitter.com/" + twitterHandle
+      + "?ref_src=twsrc%5Etfw";
+  scope.$apply();
+  const socialMediaContainer = document.getElementById("social-media-container");
+  const twitterScript = document.createElement('script');
+  twitterScript.src = "https://platform.twitter.com/widgets.js"
+  socialMediaContainer.appendChild(twitterScript);
+}
```
Comment (KathyFeiyang) [inline project/src/main/webapp/candidate-page/candidate-page.component.js:68]: nit: add new line

File: project/src/main/webapp/candidate-page/candidate-page.template.html [hunk 3]
```diff
@@ -1,36 +1,34 @@
...
-        dwellers have sent 1,000+ hate messages to my personal email,
-        website, & social accounts. Yesterday, the day my campaign
-        commercial launched, they convinced twitter to suspend my
-        account & stop my ad from going viral. Please visit my page and
-        RT my ad!"/>
-  </span>
-</div>
+  <a class="twitter-timeline" data-width="400" data-height="600"
+      href={{twitterURL}}>Tweets by {{twitterHandle}}</a>
+</div>
```
Comment (KathyFeiyang) [inline project/src/main/webapp/candidate-page/candidate-page.template.html]: nit: add new line

File: project/src/main/webapp/script.js [hunk 4]
```diff
@@ -148,102 +148,4 @@ function constructCandidateTableRow(
...
-         <hr>`;
-    newsArticlesContainer.appendChild(articleElement);
-  }
-}
-
-/**
- * @TODO [Adds the social media component to the candidate page.]
- */
-function addSocialMedia(candidateID) {}
+}
```
Comment (KathyFeiyang) [inline project/src/main/webapp/script.js]: nit: add new line

File: project/src/main/webapp/candidate-page/candidate-page.template.html (hunk 3 above)
Comment (kellyktan) [inline project/src/main/webapp/candidate-page/candidate-page.template.html]: newlines

**Review by kellyktan — APPROVED** [review]

**Review by KathyFeiyang — CHANGES_REQUESTED** [review]

File: project/src/main/webapp/candidate-page/candidate-page.template.html (hunk 3 above)
Comment (ganewman) [reply project/src/main/webapp/candidate-page/candidate-page.template.html]: Oops! Every time...

**Review by KathyFeiyang — APPROVED** [review]
LGTM!

File: project/src/main/webapp/candidate-page/candidate-page.component.js [hunk 5]
```diff
@@ -17,9 +17,52 @@
...
+  const response =
+      await fetch(
+          `/candidate?candidateId=${candidateId}&electionName=${electionName}`);
+  const dataPackage = await response.json();
+
+  // Unpack response.
+  const officialCandidateInfo = dataPackage.candidateData;
+  const newsArticles = dataPackage.newsArticlesData;
+
+  // Add 2 components of information to the scope.
```
Comment (suparnabarua) [inline project/src/main/webapp/candidate-page/candidate-page.component.js:45]: nit. Remove redundant comment

**Review by suparnabarua — APPROVED** [review]

**Review by tjyoung99 — APPROVED** [review]

**Pull request merged**